minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloConfigImpl.java

```diff
@@ -812,4 +820,29 @@ public int getNumCompactors() {
   public void setNumCompactors(int numCompactors) {
     this.numCompactors = numCompactors;
   }
+
+  /**
+   * Set the FinalSiteConfigUpdater instance that will be used to modify the site configuration
+   * right before it's written out a file. This would be useful in the case where the configuration
+   * needs to be updated based on a property that is set in MiniAccumuloClusterImpl like
+   * instance.volumes
+   *
+   * @param updater FinalSiteConfigUpdater instance
+   * @since 2.1.1
+   */
+  public void setFinalSiteConfigUpdater(FinalSiteConfigUpdater updater) {
```

Comment:
Even if the parameter is changed to a Consumer, this method still needs a decent name. I still have reservations about this one. One thing it has going for it is that it's not public API... it's only on the Impl class.

Still, maybe something like:
```suggestion
  public void preStartProcessor(Consumer<MiniAccumuloConfigImpl> prestartConfigProcessor) {
```